taxi/tests/tests_search.py

```diff
@@ -0,0 +1,174 @@
+        self.client.login(
+            username="testuser",
+            password="testpass"
+        )
+
+    def test_manufacturer_search_results(self):
+        response = self.client.get(reverse(
+            "taxi:manufacturer-list"),
+            {"manufacturer": "Ford"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.context["manufacturer_list"]), 1)
+        self.assertEqual(
+            response.context["manufacturer_list"][0].name,
+            "Ford"
+        )
+
+    def test_manufacturer_search_no_results(self):
+        response = self.client.get(reverse(
+            "taxi:manufacturer-list"),
+            {"manufacturer": "Honda"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            len(response.context["manufacturer_list"]), 0
+        )
+
+    def test_manufacturer_search_partial_match(self):
+        response = self.client.get(
+            reverse("taxi:manufacturer-list"),
+            {"manufacturer": "T"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            len(response.context["manufacturer_list"]), 2
+        )
+        manufacturers = [
+            manufacturer.name for manufacturer
+            in response.context["manufacturer_list"]
+        ]
+        self.assertIn("Toyota", manufacturers)
+        self.assertIn("Tesla", manufacturers)
+
+
+class CarSearchViewTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = User.objects.create_user(
+            username="testuser", password="testpass"
+        )
+        manufacturer = Manufacturer.objects.create(
+            name="Toyota", country="Japan"
+        )
+
+        Car.objects.create(model="Camry", manufacturer=manufacturer)
+        Car.objects.create(model="Corolla", manufacturer=manufacturer)
+        Car.objects.create(model="Civic", manufacturer=manufacturer)
+
+    def setUp(self):
+        self.client.login(
+            username="testuser",
+            password="testpass"
+        )
+
+    def test_car_search_results(self):
+        response = self.client.get(
+            reverse("taxi:car-list"), {"car": "Camry"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.context["car_list"]), 1)
+        self.assertEqual(
+            response.context["car_list"][0].model, "Camry"
+        )
+
+    def test_car_search_no_results(self):
+        response = self.client.get(
+            reverse("taxi:car-list"), {"car": "Accord"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.context["car_list"]), 0)
+
+    def test_car_search_partial_match(self):
+        response = self.client.get(
+            reverse("taxi:car-list"), {"car": "C"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.context["car_list"]), 3)
```

Comment:
Ensure that the car search view logic supports partial matches, as this test assumes partial matching functionality.